plugins/modules/win_domain_ou.ps1

```diff
@@ -0,0 +1,257 @@
+                break
+            }
+            $Properties.Add($_)
+        })
+        if ($changed){
+            return $true
+        }
+        $x = Compare-Object -ReferenceObject $module.diff.before -DifferenceObject $module.diff.after -Property $Properties
+        if ($x.Count -ne 0){
+            return $true
+        }
+        # $module.FailJson("Found Properties by count greater than 0 but did not find a change count: $($module.Params.properties.Count) keys: $($module.Params.properties.keys)")
+    }
+    # check for difference when properties is *
+    $x = Compare-Object -ReferenceObject $module.diff.before -DifferenceObject $module.diff.after
+    if ($x.Count -ne 0){
+      return $true
+    }
+    return $false
+}
+
+Function Get-SimulatedOu {
+    Param($Object)
+    $params = @{
+        Name = $Object.name
+        DistinguishedName = "OU=$($Object.name),$($Object.path)"
+        ProtectedFromAccidentalDeletion = $Object.protected
+    }
+    if ($Object.properties) {
+        if ($Object.properties.description) { $params.Description = $Object.properties.description }
+        if ($Object.properties.city) { $params.City = $Object.properties.city }
+        if ($Object.properties.state) { $params.State = $Object.properties.state }
+        if ($Object.properties.street_address) { $params.StreetAddress = $Object.properties.street_address }
+        if ($Object.properties.postal_code) { $params.PostalCode = $Object.properties.postal_code }
+        if ($Object.properties.country) { $params.Country = $Object.properties.country }
+        if ($Object.properties.managed_by) { $params.ManagedBy = $Object.properties.managed_by }
+    }
+    # convert to psobject & return
+    [PSCustomObject]$params
+}
+
+Function Get-OuObject {
+    Param([PSObject]$Object)
+    $obj = $Object | Select-Object -Property * -ExcludeProperty nTSecurityDescriptor | ConvertTo-Json -Depth 1 | ConvertFrom-Json
+    return $obj
+}
+
+# attempt import of module
+Try { Import-Module ActiveDirectory }
+Catch { $module.FailJson("The ActiveDirectory module failed to load properly: $($_.Exception.Message)", $_) }
+
+Try{
+    $all_ous = Get-ADOrganizationalUnit @extra_args
+}Catch{$module.FailJson("Get-ADOrganizationalUnit failed: $($_.Exception.Message)", $_) }
+
+# set path if not defined to base domain
+if ($null -eq $path){
+    if ($($all_ous | Measure-Object | Select-Object -ExpandProperty Count) -eq 1){
+        $matched = $all_ous.DistinguishedName -match "DC=.+"
+    }elseif ($($all_ous | Measure-Object | Select-Object -ExpandProperty Count) -gt 1) {
+        $matched = $all_ous[0].DistinguishedName -match "DC=.+"
+    }else{
+        $module.FailJson("Path was null and unable to determine default domain $($_.Exception.Message)", $_)
+    }
+    if ($matched){
+        $path = $matches.Values[0]
+    }else{
+        $module.FailJson("Unable to find default domain $($_.Exception.Message)", $_)
+    }
+}
+$module.Result.path = $path
```

Comment:
This needs to be document in the module docs, currently there's only `ou`.

Reply:
added to document